scalafix-cli/src/main/scala/scalafix/internal/jgit/JGitDiff.scala

```diff
@@ -0,0 +1,103 @@
+package scalafix.internal.jgit
+
+import java.nio.file.Path
+
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder
+import org.eclipse.jgit.api.Git
+import org.eclipse.jgit.api.errors.GitAPIException
+import org.eclipse.jgit.diff.DiffEntry.ChangeType._
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.patch.FileHeader
+import org.eclipse.jgit.lib.ObjectReader
+import org.eclipse.jgit.lib.Ref
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.ProgressMonitor
+import org.eclipse.jgit.dircache.DirCacheIterator
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevTree
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.AbstractTreeIterator
+import org.eclipse.jgit.treewalk.CanonicalTreeParser
+import org.eclipse.jgit.treewalk.FileTreeIterator
+import org.eclipse.jgit.util.io.NullOutputStream
+import scala.collection.JavaConverters._
+
+object JGitDiff {
+  def apply(workingDir: Path, baseBranch: String): List[GitDiff] = {
+    val builder = new FileRepositoryBuilder()
+    val repository =
+      builder.readEnvironment().setWorkTree(workingDir.toFile).build()
+
+    val baseBranchRef = s"refs/heads/$baseBranch"
+    val git = new Git(repository)
+    val branches = git.branchList().call()
+    if (branches.isEmpty) {
+      throw new Exception(s"$workingDir is not a git repository")
+    }
+    val baseBranchExists = branches.asScala.exists(_.getName() == baseBranchRef)
+    if (!baseBranchExists) {
+      throw new Exception(s"Cannot find git branch $baseBranch")
```

Comment:
This error is reported for tags, can we treat tags like branches?

```
scalafix-nightly -r DisableSyntax --config-str="DisableSyntax.keywords=[def]" --diff-branch=v0.5.6                                                                                        8   drop-in-jgit
error: java.lang.Exception: Cannot find git branch v0.5.6
scalafix.internal.jgit.JGitDiff$.apply(JGitDiff.scala:40)
```

Reply:
I will rename --diff-branch= to --diff-base= and allow more options (hash and tags)